src/NuGet.Clients/PackageManagement.UI/Xamls/InfiniteScrollList.xaml.cs

```diff
@@ -77,17 +79,9 @@ public bool CheckBoxesEnabled
 
         public bool IsSolution { get; set; }
 
-        private readonly ObservableCollection<object> _items = new ObservableCollection<object>(); 
-
-        public ObservableCollection<object> Items
-        {
-            get
-            {
-                Debug.Assert(Mvs.ThreadHelper.CheckAccess());
-                return _items;
-            }
-        }
+        private object _itemsLock = new object();
 
+        public ObservableCollection<object> Items { get; } = new ObservableCollection<object>();
         public IEnumerable<PackageItemListViewModel> PackageItems => Items.OfType<PackageItemListViewModel>().ToArray();
```

Comment:
now that i think of it, wouldn't calling ToArray() make PackageItems and Items to be out of sync?